src/Manifolds.jl

```diff
@@ -89,6 +89,12 @@ using Random: AbstractRNG
 using UnsafeArrays
 using Einsum: @einsum
 
+abstract type AbstractMethod end
```

Comment:
Could we perhaps have a less generic name for this type? Or maybe a docstring which explains what kinds of methods it covers?

Reply:
Yes this is not ideal. I was thinking something like `AbstractMeanMethod` but e.g. `GradientMethod` is a generic name that could be useful for other functions. I don't know about `CyclicProximalPointMethod` though.

Reply:
`CyclicProximalPointMethod` is generic in the sense that is just describes the optimization approach like `GradientMethod`.

Reply:
What if we introduced a new `AbstractMethod` in ManifoldsBase that is the parent type for our other methods? Then we can have a subtype AbstractOptimizationMethod that these inherit from.

Reply:
I think I prefer to just have `AbstractOptimizationMethod` and `AbstractEstimationMethod` here. They don't need a common base type, do they?